src/components/common/field-selector.js

```diff
@@ -24,33 +24,36 @@ import PropTypes from 'prop-types';
 import {createSelector} from 'reselect';
 
 import ItemSelector from './item-selector/item-selector';
-import FieldToken from '../common/field-token';
 import {classList} from './item-selector/dropdown-list';
 import {toArray} from 'utils/utils';
 import {notNullorUndefined} from 'utils/data-utils';
+import FieldTokenFactory from '../common/field-token';
 
 const defaultDisplayOption = d => d.name;
 
 const StyledToken = styled.div`
   display: inline-block;
-  margin: 0 4px 0 0;
+  margin: 0 ${props => props.theme.tokenRightMargin}px 0 0;
 `;
 
+FieldListItemFactoryFactory.deps = [FieldTokenFactory];
 // custom list Item
-export const FieldListItemFactory = (showToken = true) => {
-  const FieldListItem = ({value, displayOption = defaultDisplayOption}) => (
-    <div className="field-selector_list-item">
-      {showToken ? (
-        <StyledToken>
-          <FieldToken type={value.type} />
-        </StyledToken>
-      ) : null}
-      <span className={classList.listItemAnchor}>{displayOption(value)}</span>
-    </div>
-  );
-
-  return FieldListItem;
-};
+export function FieldListItemFactoryFactory(FieldToken) {
```

Comment:
`FieldListItemFactoryFactory` -> `FieldListItemFactory`?

Reply:
the previous component had already "Factory" in its name, but wasn't a kepler factory